src/miral/wayland_extensions.cpp

```diff
@@ -93,6 +98,30 @@ struct StaticExtensionTracker
 
 decltype(StaticExtensionTracker::mutex)       StaticExtensionTracker::mutex;
 decltype(StaticExtensionTracker::extensions)  StaticExtensionTracker::extensions;
+
+std::map<std::string, std::string> const alternative_extension_names{
+    {"zwp_input_method_v2", "zwp_input_method_manager_v2"},
+    {"zwp_virtual_keyboard_v1", "zwp_virtual_keyboard_manager_v1"},
+};
```

Comment:
[Non blocking]

I'm probably old fashioned, but it seems odd to use `map<>` for such small N. A couple of `if ... else if ...` conditions would avoid all the non-constinit setup and teardown.